bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -1,22 +1,39 @@
 import { useAccount } from "wagmi";
 import { useQuery } from "@tanstack/react-query";
-import { useChainStore } from "@/stores";
-import { fetchTransactionsHistory } from "@/utils";
+import { HistoryActionsForCompleteTxCaching, useChainStore } from "@/stores";
 import useLineaSDK from "./useLineaSDK";
 import useTokens from "./useTokens";
+import { useHistoryStore } from "@/stores";
+import { fetchTransactionsHistory } from "@/utils";
 
 const useTransactionHistory = () => {
   const { address } = useAccount();
   const fromChain = useChainStore.useFromChain();
   const toChain = useChainStore.useToChain();
-  const { lineaSDK, lineaSDKContracts } = useLineaSDK();
+  const { lineaSDK } = useLineaSDK();
   const tokens = useTokens();
+  const { setCompleteTx, getCompleteTx } = useHistoryStore((state) => ({
+    setCompleteTx: state.setCompleteTx,
+    getCompleteTx: state.getCompleteTx,
+  }));
+  const historyStoreActions: HistoryActionsForCompleteTxCaching = {
+    setCompleteTx,
+    getCompleteTx,
+  };
 
+  // TODO - For invalidateQueries with queryKey: "transactionHistory", we don't need to refetch the entire transaction history, just the single affected transaction
```

Comment:
when will this be done?

Reply:
well, this requires a re-design of the components and the data flow.

I will remove the comment, as after diving deeper into the code I don't have a good suggestion for what to do here atm